src/simulation/in_memory_simulation_store.jl

```diff
@@ -64,88 +59,97 @@ function write_optimizer_stats!(
     stats::OptimizerStats,
     timestamp::Dates.DateTime,
 )
-    store.optimizer_stats[Symbol(model_name)][timestamp] = stats
+    write_optimizer_stats!(store.dm_data[model_name], stats, timestamp)
+    return
+end
+
+function write_optimizer_stats!(
+    store::InMemorySimulationStore,
+    model_name,
+    stats::OptimizerStats,
+    execution::Int,
+)
+    write_optimizer_stats!(store.em_data[model_name], stats, index)
     return
 end
 
 function read_model_optimizer_stats(
     store::InMemorySimulationStore,
-    ::Int,
     model_name,
     timestamp::Dates.DateTime,
 )
-    _check_timestamp(store.optimizer_stats, timestamp)
-    return store.optimizer_stats[model_name][timestamp]
+    return read_optimizer_stats(store.dm_data[model_name], timestamp)
 end
 
-function read_model_optimizer_stats(store::InMemorySimulationStore, model_name)
-    stats = [to_namedtuple(x) for x in values(store.optimizer_stats[model_name])]
-    return DataFrames.DataFrame(stats)
+function read_model_optimizer_stats(
+    store::InMemorySimulationStore,
+    model_name,
+    execution::Int,
+)
+    return read_optimizer_stats(store.em_data[model_name], execution)
 end
 
-function initialize_model_storage!(
+function initialize_problem_storage!(
     store::InMemorySimulationStore,
     params,
-    model_reqs,
+    problem_reqs,
     flush_rules,
 )
     store.params = params
-    @debug "initialize in memory storage"
-
-    for model_name in keys(store.params.models)
-        store.data[model_name] = DecisionModelOptimizerResults()
+    for problem in keys(store.params.models_params)
+        store.dm_data[problem] = DecisionModelOptimizerResults()
         for type in STORE_CONTAINERS
-            for (name, reqs) in getfield(model_reqs[model_name], type)
-                container = getfield(store.data[model_name], type)
+            for (name, reqs) in getfield(problem_reqs[problem], type)
+                container = getfield(store.dm_data[problem], type)
                 container[name] = OrderedDict{Dates.DateTime, DataFrames.DataFrame}()
+                @debug "Added $type $name in $problem" _group = LOG_GROUP_SIMULATION_STORE
             end
         end
 
-        store.optimizer_stats[model_name] = OrderedDict{Dates.DateTime, OptimizerStats}()
-        @debug "Initialized optimizer_stats_datasets $model_name"
+        # TODO EmulationModel: how do we differentiate DM and EM?
```

Comment:
Is this something we can fix now?

Reply:
I was thinking about using `get_decision_model_results` and `get_emulation_model_results` to emphasize the difference. Honing down the concept on users is critical.

Reply:
You might have missed my question. In this function we iterate over the problems in `problem_reqs`. It is currently assuming that every problem is a DecisionModel. If there is an EmulationModel, how do we tell?

Reply:
> You might have missed my question. In this function we iterate over the problems in `problem_reqs`. It is currently assuming that every problem is a DecisionModel. If there is an EmulationModel, how do we tell?

TBD, the trick here is that for 1.0 we will support only one EM and if there is none, this is populated from the `SystemState`

Reply:
The HDF5 datasets need to get created before the simulation starts. How will that happen?

Reply:
> The HDF5 datasets need to get created before the simulation starts. How will that happen?

We know how many executions will happen the tables will have `n_executions` number of rows.